data_prep/introspector.py

```diff
@@ -57,6 +61,86 @@ def query_introspector_cfg(project):
   return data.get('project', {})
 
 
+def query_introspector_function_source(project: str, func_sig: str) -> str:
+  """Queries FuzzIntrospector API for source code of |func_sig| in |project|."""
+  resp = requests.get(INTROSPECTOR_SOURCE,
+                      params={
+                          'project': project,
+                          'function_signature': func_sig
+                      },
+                      timeout=TIMEOUT)
+  data = resp.json()
+  source = data.get('source', '')
+  if not source:
+    logging.error(
+        f'No function source found for {func_sig} in {project}: {data}')
+
+  return source
+
+
+def query_introspector_cross_references(project: str,
+                                        func_sig: str) -> list[str]:
+  """Queries FuzzIntrospector API for source code of functions cross-referenced |func_sig| in |project|."""
+  resp = requests.get(INTROSPECTOR_XREF,
+                      params={
+                          'project': project,
+                          'function_signature': func_sig
+                      },
+                      timeout=TIMEOUT)
+  if not resp.ok:
+    logging.error(f'Failed to get cross references '
+                  f'for function: {func_sig} in project: {project}\n'
+                  f'-----------Response received------------\n'
+                  f'{resp.content.decode("utf-8").strip()}\n'
+                  f'------------End of response-------------')
+    return []
+
+  data = resp.json()
```

Comment:
These steps (request, failure handling, and jsonify) repeat in multiple functions.
Could you make it a function to avoid duplication?

Reply:
done